lib/gen_stage.ex

```diff
@@ -974,6 +976,14 @@ defmodule GenStage do
               | {:stop, reason, new_state}
             when new_state: term, reason: term
 
+  @doc """
+  Invoked when items are discarded from the buffer.
+
+  It receives the number of excess (discarded) items from this invocation.
+  When this callback is implemented, the error log for discarded items is bypassed so that the callback can handle it.
+  """
+  @callback handle_discarded(discarded :: non_neg_integer, state :: term) :: none
```

Comment:
Not a big fan of the `handle_discarded` name, since it is not an usual callback. There is a similar-ish callback in GenServer that they call `format_status` but I am not sure if that's better.

Reply:
I am new to naming conventions in elixir and open to ideas.
I modeled the name of this one after the other GenStage callbacks.
GenStage also has a `format_status` callback that closely matches GenServer's.

Reply:
What about `format_discarded` which returns a boolean if GenStage should log or not?

Reply:
That sounds good to me, I'll do that.